src/ledger-keyring.ts

```diff
@@ -308,27 +199,22 @@ export class LedgerBridgeKeyring extends EventEmitter {
       return 'already unlocked';
     }
     const path = hdPath ? this.#toLedgerPath(hdPath) : this.hdPath;
-    return new Promise((resolve, reject) => {
-      this.#sendMessage(
-        {
-          action: IFrameMessageAction.LedgerUnlock,
-          params: {
-            hdPath: path,
-          },
-        },
-        ({ success, payload }) => {
-          if (success && isGetAddressMessageResponse(payload)) {
-            if (updateHdk) {
-              this.hdk.publicKey = Buffer.from(payload.publicKey, 'hex');
-              this.hdk.chainCode = Buffer.from(payload.chainCode, 'hex');
-            }
-            resolve(payload.address);
-          } else {
-            reject(payload.error ?? new Error('Unknown error'));
-          }
-        },
-      );
-    });
+
+    let payload;
+    try {
+      payload = await this.bridge.getPublicKey({
+        hdPath: path,
+      });
+    } catch (error) {
+      throw error instanceof Error ? error : new Error('Unknown error');
+    }
+
+    if (updateHdk && payload.chainCode) {
+      this.hdk.publicKey = Buffer.from(payload.publicKey, 'hex');
```

Comment:
we have changed the `payload.publicKey` type checking from runtime to build time. Just want to make sure that that is enough.

Reply:
Should be if we trust the return types from ledger SDK.

Before, multiple checks where needed due to `#sendMessage` using an `IFrameMessageResponse` that would not consider the action, and therefore return just an union type. Now we have a more accurate `IFrameMessageResponse` that is based on the action sent to ledger.